elm/web/rhub.py

```diff
@@ -0,0 +1,646 @@
+                                                  'url', 'doi',
+                                                  'pdf_url', 'category'))
+        for link in self.all_links[:20]:  # quantity control here #
+            with urlopen(link) as page:
+                html = page.read().decode("utf-8")
+            meta_soup = BeautifulSoup(html, "html.parser")
+
+            title = meta_soup.find('h1').text
+            nrel_id = self._scrape_id(meta_soup)
+            authors = self._scrape_authors(meta_soup)
+            doi = self._scrape_links(meta_soup)[0]
+            pdf_url = self._scrape_links(meta_soup)[1]
+            category = self._scrape_category(meta_soup)
+            year = self._scrape_year(meta_soup)
+
+            new_row = {'title': title,
+                       'nrel_id': nrel_id,
+                       'year': year,
+                       'authors': authors,
+                       'url': link,
+                       'doi': doi,
+                       'pdf_url': pdf_url,
+                       'category': category
+                       }
+
+            publications_meta.loc[len(publications_meta)] = new_row
+
+        return publications_meta
+
+    def download_pdf(self, pdf_dir, txt_dir, soup_inst):
+        """Downloads a pdf for a given link
+
+        Parameters
+        ----------
+        out_dir: str
+            Directory where the .pdf files should be saved.
+        soup_inst : bs4.BeautifulSoup
+            Active beautiful soup instance used to locate pdf url.
+        """
+        pdf_target = soup_inst.find('ul', {'class': 'links'})
+        if pdf_target:
+            pdf_url = pdf_target.find('a')['href']
+
+        fn = os.path.basename(pdf_url)
+        fp_out = os.path.join(pdf_dir, fn)
+
+        if pdf_url and pdf_url.endswith('.pdf'):
+            if not os.path.exists(fp_out):
+                session = requests.Session()
+                response = session.get(pdf_url)
+                with open(fp_out, 'wb') as f_pdf:
+                    f_pdf.write(response.content)
+                logger.info('Downloaded {}'.format(fn))
+            else:
+                logger.info('{} has already been downloaded'.format(fn))
+        elif not pdf_url.endswith('.pdf'):
+            parent_url = soup_inst.find(property="og:url")['content']
+            fn = os.path.basename(parent_url) + '_abstract.txt'
+            logger.info('No PDF file for {}. Processing abstract.'.format(fn))
+            self.scrape_abstract(txt_dir, fn, soup_inst)
+
+    def scrape_abstract(self, out_dir, fn, soup_inst):
+        """
+        Description
+        ----------
+        Scrapes abstract for a provided publication
+
+        Parameters
+        ----------
+        out_dir: str
+            Directory where the .txt files should be saved.
+        fn: str
+            File name for saving the file.
+        soup_inst : bs4.BeautifulSoup
+            Active beautiful soup instance used for scraping.
```

Comment:
The correct term is "instantiated" not "active".  Recommended fixing everywhere.